### ISSUE TITLE
Bug in 1d

### ISSUE THREAD

**Issue by Binbose** [ChrisWaites/jax-flows#9]
I am trying to fit a simple toy example in 1d, from different papers I have seen that it should be working, but the model doesn't learn the modes at all, it essentially only stays a gaussian.

![Figure_1](https://user-images.githubusercontent.com/25252778/164052886-a9481857-ae82-47b2-bace-2671951e86ad.png)

I tried different loss functions, either the standard -log(sample), but also directly KL divergence between the known groundtruth likelihood and the model likelihood.
Am I doing something wrong or is there a Bug for 1d (the 2d case from the half-moon examples works fine)
Here is the test code

```
import numpy as np
import matplotlib.pyplot as plt
import jax.numpy as jnp
import flows
from jax import grad, jit, random
import itertools
from jax.example_libraries import stax, optimizers
import tqdm
from sklearn import datasets, preprocessing, mixture


# jax.config.update('jax_platform_name', 'cpu')

length = 4

x = jnp.linspace(-length/2, length/2, 1000)[:,None]
input_dim = 1
split_key, rng = random.split(random.PRNGKey(0))

gmm = mixture.GaussianMixture(3)
means = jnp.array([[ 0.4570112 ], [-0.97355632], [-0.18146387]])
covariances = jnp.array([[[0.0650186]], [[0.00751129]], [[0.0501704 ]]])
weights = jnp.array([0.32999969, 0.42999999, 0.34000032])
init_fun = flows.GMM(means, covariances, weights)

params_gt, log_pdf_gt, sample_gt = init_fun(split_key, input_dim)
y = np.exp(log_pdf_gt(params_gt, x))


plt.plot(x, y, label='Density')
plt.legend()
plt.show()

def get_masks(input_dim, hidden_dim=64, num_hidden=1):
    masks = []
    input_degrees = jnp.arange(input_dim)
    degrees = [input_degrees]

    for n_h in range(num_hidden + 1):
        degrees += [jnp.arange(hidden_dim) % (input_dim - 1)]
    degrees += [input_degrees % input_dim - 1]

    for (d0, d1) in zip(degrees[:-1], degrees[1:]):
        masks += [jnp.transpose(jnp.expand_dims(d1, -1) >= jnp.expand_dims(d0, 0)).astype(jnp.float32)]
    return masks


def masked_transform(rng, input_dim):
    masks = get_masks(input_dim, hidden_dim=64, num_hidden=1)
    # act = stax.Relu
    act = stax.Selu
    init_fun, apply_fun = stax.serial(
        flows.MaskedDense(masks[0]),
        act,
        flows.MaskedDense(masks[1]),
        act,
        flows.MaskedDense(masks[2].tile(2)),
    )
    _, params = init_fun(rng, (input_dim,))
    return params, apply_fun


bijection = flows.Serial(*(flows.MADE(masked_transform), flows.Reverse()) * 5)
params, direct_fun, inverse_fun = bijection(random.PRNGKey(0), x.shape[-1])
init_fun = flows.Flow(bijection, flows.Normal())
input_dim = x.shape[1]
num_epochs, batch_size = 2100000, 1000

params, log_pdf, sample = init_fun(split_key, input_dim)

opt_init, opt_update, get_params = optimizers.adam(step_size=1e-4)
opt_state = opt_init(params)
def loss(params, inputs):
    # groundtruth = jnp.exp(log_pdf_gt(params_gt, inputs))
    # log_groundtruth = jnp.log(groundtruth)
    # weight = jax.lax.stop_gradient(jnp.exp(log_pdf(params, inputs)))

    # return = ((log_pdf(params, inputs)[:,None] - jnp.log(ground_truth))**2).mean()
    # return ((jnp.exp(log_pdf(params, inputs))[:,None] - ground_truth)**2).mean()
    # loss_val = ((groundtruth.squeeze() - jnp.exp(log_pdf(params, inputs)))**2).mean()

    loss_val = -log_pdf(params, inputs).mean()

    return loss_val


@jit
def step(i, opt_state, inputs):
    params = get_params(opt_state)
    loss_val = loss(params, inputs)
    gradients = grad(loss)(params, inputs)
    return opt_update(i, gradients, opt_state), loss_val


itercount = itertools.count()
pbar = tqdm.tqdm(range(num_epochs))

params = get_params(opt_state)
for epoch in pbar:
    losses = []
    if epoch % 5000 == 0 and epoch > 0:
        params = get_params(opt_state)
        sample_rng, rng = random.split(rng)
        flow_pdf = jnp.exp(log_pdf(params, x))
        plt.plot(x, flow_pdf, label='Model')
        plt.plot(x, jnp.exp(log_pdf_gt(params_gt, x)), label='GT')
        plt.legend()
        plt.show()

    split_rng, rng = random.split(rng)
    # X = sample(rng, params, 1000)
    # X = random.permutation(permute_rng, x)
    # Y = random.permutation(permute_rng, y2)
    # X = jax.random.uniform(split_rng, (batch_size, 1), minval=-length//2, maxval=length//2)
    X = sample_gt(split_rng, params_gt, batch_size)[:,None]
    opt_state, loss_val = step(epoch, opt_state, X)
    losses.append(loss_val)

    pbar.set_description('Loss {}'.format(jnp.array(losses).mean()))

```
